network/src/transport.rs

```diff
@@ -281,25 +281,6 @@ pub fn build_unauthenticated_memory_noise_transport(
         .boxed()
 }
 
-pub fn build_memory_transport(
```

Comment:
do we want to remove this? it gives us insights into performance? though I guess it adds tech debt?

Reply:
socket bench is just using a `MemoryTransport` directly, so we don't really need these anymore